crates/kornia-imgproc/src/features.rs

```diff
@@ -68,6 +82,51 @@ pub fn hessian_response(src: &Image<f32, 1>, dst: &mut Image<f32, 1>) -> Result<
     Ok(())
 }
 
+/// Compute the DoG response of an image.
+///
+/// The DoG response is computed as the difference of the Gaussian responses of two images.
+///
+/// Args:
+///     src: The source image with shape (H, W).
+///     dst: The destination image with shape (H, W).
+///     sigma1: The sigma of the first Gaussian kernel.
+///     sigma2: The sigma of the second Gaussian kernel.
+pub fn dog_response_single(
+    src: &Image<f32, 1>,
+    dst: &mut Image<f32, 1>,
+    sigma1: f32,
+    sigma2: f32,
+) -> Result<(), ImageError> {
+    if src.size() != dst.size() {
+        return Err(ImageError::InvalidImageSize(
+            src.cols(),
+            src.rows(),
+            dst.cols(),
+            dst.rows(),
+        ));
+    }
+
+    let mut gauss1 = Image::from_size_val(src.size(), 0.0)?;
+    let mut gauss2 = Image::from_size_val(src.size(), 0.0)?;
```

Comment:
we can think of that later but probably to make really efficient this operator we might want to implement it as a struct and allocate only once the this temporary variables